hs/src/Reach/Optimize.hs

```diff
@@ -447,16 +446,7 @@ class Extract a where
 instance Extract (Maybe DLVar) where
   extract = id
 
-allTheSame :: (Eq a, Sanitize a) => [a] -> Maybe a
-allTheSame xs =
-  case map sani xs of
```

Comment:
Notice that my version uses `Sanitize` to remove SrcLocs and that makes it more likely things will be equal. If you want to replace `allTheSame`, then either you should use `Sanitize` too or you should do `allTheSame = allEqual . map sani`